src/plotman/job.py

```diff
@@ -137,28 +137,45 @@ def get_running_jobs(logroot, cached_jobs=()):
         jobs = []
         cached_jobs_by_pid = { j.proc.pid: j for j in cached_jobs }
 
-        for proc in psutil.process_iter():
-            # Ignore processes which most likely have terminated between the time of
-            # iteration and data access.
-            with contextlib.suppress(psutil.NoSuchProcess, psutil.AccessDenied):
-                if is_plotting_cmdline(proc.cmdline()):
-                    if proc.pid in cached_jobs_by_pid.keys():
-                        jobs.append(cached_jobs_by_pid[proc.pid])  # Copy from cache
-                    else:
-                        with proc.oneshot():
-                            parsed_command = parse_chia_plots_create_command_line(
-                                command_line=proc.cmdline(),
-                            )
-                            if parsed_command.error is not None:
-                                continue
-                            job = Job(
-                                proc=proc,
-                                parsed_command=parsed_command,
-                                logroot=logroot,
-                            )
-                            if job.help:
-                                continue
-                            jobs.append(job)
+        with contextlib.ExitStack() as exit_stack:
+            processes = []
+
+            for process in psutil.process_iter():
+                # Ignore processes which most likely have terminated between the time of
+                # iteration and data access.
+                with contextlib.suppress(psutil.NoSuchProcess, psutil.AccessDenied):
+                    exit_stack.enter_context(process.oneshot())
+                    if is_plotting_cmdline(process.cmdline()):
+                        processes.append(process)
+
+            pids = {process.pid for process in processes}
+            ppids = {process.ppid() for process in processes}
+            wanted_pids = pids - ppids
+
+            wanted_processes = [
+                process
+                for process in processes
+                if process.pid in wanted_pids
+            ]
```

Comment:
A nit but a good nit: add a small comment saying why this is necessary, maybe even reference this PR, as it may not be clear the reason this is necessary

Reply:
Thanks, added.  Does it read well?

Reply:
Superb 👨‍🍳 💋